enhancements/workload-partitioning/wide-availability-workload-partitioning.md

```diff
@@ -0,0 +1,731 @@
+     Node, will check with old logic to maintain upgrade for Single-Node 4.11 ->
+     4.12.
+3. Checks what resource limits and requests are set on the pod
+   - Skips modification if QoS is guaranteed or both limits and requests are set
+   - Skips modification if after update the QoS is changed
+
+### Risks and Mitigations
+
+The sames risks and mitigations highlighted in [Management Workload
+Partitioning](management-workload-partitioning.md) apply to this enhancement as
+well.
+
+We need to make it very clear to customers that this feature is supported as a
+day 0 configuration and day n+1 alterations are not be supported with this
+enhancement. Part of that messaging should involve a clear indication that this
+currently will be a cluster wide feature.
+
+A risk we can run into is that a customer can apply a CPU set that is too small
+or out of bounds can cause problems such as extremely poor performance or start
+up errors. Mitigation of this scenario will be to provide proper guidance and
+guidelines for customers who enable this enhancement. As mentioned in our goal
+we do support re-configuring the CPUSet partition size after installation.
+
+Another similar risk to CPU set sizes being too small or out of bounds is if
+customers mix nodes of differing CPU set sizes in the same machine config pool.
+This is not supported with this feature as it is required that machine pools
+contain machines of the same size to correctly apply the CPUSet affinities. If
+customers mix node sizes in the same pool, then the ones where the CPUSet is out
+of bound will fail, while the others will not. They will need to evict those
+machines and add ones that fall with in bounds of the `reserved` and `isolated`
+CPUSets defined in their PerformanceProfile.
+
+A possible risk are cluster upgrades, this is the first time this enhancement
+will be for multi-node clusters, we need to run more tests on upgrade cycles to
+make sure things run as expected.
+
+### Drawbacks
+
+This feature contains the same drawbacks as the [Management Workload
+Partitioning](management-workload-partitioning.md).
+
+Several of the changes described above are patches that we may end up carrying
+downstream indefinitely. Some version of a more general "CPU pool" feature may
+be acceptable upstream, and we could reimplement management workload
+partitioning to use that new implementation.
+
+## Design Details
+
+### Open Questions [optional]
+
+N/A
+
+### Test Plan
+
+We will add a CI job with a cluster configuration that reflects the minimum of
+2CPU/4vCPU masters and 1CPU/2vCPU worker configuration. This job should ensure
+that cluster deployments configured with management workload partitioning pass
+the compliance tests.
+
+We will add a CI job to ensure that all release payload workloads have the
+`target.workload.openshift.io/management` annotation and their namespaces have
+the `workload.openshift.io/allowed` annotation.
+
+### Graduation Criteria
```

Comment:
What support level will the first iteration of this have? Tech preview or GA?

Reply:
PerformanceProfile is a stable v2 API and any changes will be perceived as stable as well unless a new API revision is created (v3alpha1).